Model/Indexer/AbstractIndexer.php

```diff
@@ -0,0 +1,216 @@
+                    if ($this->isDimensionProcessable($dimension)) {
+                        /** @suppress PhanTypeMismatchArgument */
+                        $userFunctions[] = function () use ($dimension, $ids) {
+                            $this->executeByDimensions($dimension, new ArrayIterator($ids));
+                        };
+                    }
+                }
+                /** @var Traversable $userFunctions  */
+                $this->getProcessManager()->execute($userFunctions);
+                break;
+            default:
+                throw new UnexpectedValueException("Undefined dimension mode.");
+        }
+    }
+
+    /**
+     * In case processManager has value null, pass a new instance of ProcessManager
+     * This operation is not done in the constructor in order for the constr to have only
+     * value assignments
+     *
+     * @return ProcessManager
+     */
+    private function getProcessManager()
+    {
+        if ($this->processManager ===  null) {
+            $this->processManager = ObjectManager::getInstance()->get(
+                ProcessManager::class
+            );
+        }
+        return $this->processManager;
+    }
+
+    /**
+     * @param Dimension[] $dimensions
+     * @param Traversable|null $entityIds
+     * @throws NostoException
+     */
+    public function executeByDimensions(array $dimensions, Traversable $entityIds = null)
+    {
+        if (count($dimensions) > 1 || !isset($dimensions[StoreDimensionProvider::DIMENSION_NAME])) {
+            throw new InvalidArgumentException('Indexer "' . $this->getIndexerId() . '" support only Store dimension');
+        }
+
+        $storeId = $dimensions[StoreDimensionProvider::DIMENSION_NAME]->getValue();
+        $store = $this->nostoHelperScope->getStore($storeId);
+        $benchmarkName = sprintf('STORE-DIMENSION-%s', $store->getCode());
+        Benchmark::getInstance()->startInstrumentation($benchmarkName, 0);
+        $this->nostoLogger->info('[START] NOSTO-DIMENSION store:'. $store->getName());
+        $ids = [];
+        if ($entityIds !== null) {
+            $ids = iterator_to_array($entityIds);
+        }
+
+        $this->doIndex($store, $ids);
+
+        Benchmark::getInstance()->stopInstrumentation($benchmarkName);
+        $duration = Benchmark::getInstance()->getElapsed($benchmarkName);
+        $this->nostoLogger->info(
+            '[END] NOSTO-DIMENSION store:' . $store->getName() . '(' . round($duration, 2) . ' secs)'
+        );
+    }
+
+    /**
+     * @param Dimension[] $dimension
+     * @return bool
+     * @suppress PhanTypeArraySuspicious
+     */
+    private function isDimensionProcessable(array $dimension)
+    {
+        $storeId = $dimension[StoreDimensionProvider::DIMENSION_NAME]->getValue();
+        $store = $this->nostoHelperScope->getStore($storeId);
+        if ($this->nostoHelperAccount->nostoInstalledAndEnabled($store)) {
```

Comment:
Why are we doing this check in multiple places? We do it here and we also do it in call tree of the doIndex methods